core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/CompactEntityComponent.scala

```diff
@@ -208,6 +201,87 @@ class CompactEntityQuery(driverComponent: DriverComponent) extends RawSqlQuery w
       query.as[CompactEntityRefRecord]
     }
 
+  def copyEntitiesToNewWorkspace(sourceWs: UUID,
+                                 destWs: UUID,
+                                 entityRefs: Set[AttributeEntityReference] = Set()
+                                ): WriteAction[Int] = {
+
+    def copyChunkOfEntitiesOrAllEntities(chunk: Set[AttributeEntityReference] = Set()) =
+      for {
+        entitiesCopiedCount <- copyEntities(sourceWs, destWs, chunk)
+      } yield entitiesCopiedCount
+
+      val chunks: Iterator[Set[AttributeEntityReference]] = if (entityRefs.size > driverComponent.batchSize) {
+        entityRefs.grouped(driverComponent.batchSize)
+      } else {
+        Iterator(entityRefs)
+      }
+
+    val allCopies = DBIO.sequence(chunks map copyChunkOfEntitiesOrAllEntities)
```

Comment:
can you replace `copyChunkOfEntitiesOrAllEntities` with  `copyEntities(sourceWs, destWs, _)`